### DISCUSSION TITLE
What do you think about the new API?

### DISCUSSION THREAD

**Discussion by markseuffert** [datenstrom/community#401] [Ask a question] [answered]
Hi! We made a new API for develeopers and it would be great to collect more ideas. What would you like to see? Anything not working or missing that you need?

**Comment by schulle4u** [comment]
The update went smoothly. If you're getting "file not found" errors after applying the new folder structure, run another update check to fetch remaining extensions. This should solve all errors. 

The file `system/settings/system.ini` gets a little messy after this type of API changes. New options are appended at the end, and old ones are commented out. Could this be improved somehow to make configuration a little bit cleaner? Maybe a sorting algorythm that rejoins configuration lines on their proper place after update.

**Comment by markseuffert** [comment]
Good idea. I updated everything with your feedback. 

When the website is updated to the new API, it shows a message to "update your website one more time". It takes two steps to update folder structure and download remaining files. After the website is updated, system settings are ordered/grouped together and old ones are moved to the end.

**Comment by markseuffert** [comment, marked as answer]
Appears to be working, closing issue.
